resources/test/fixtures/U012.py

```diff
@@ -0,0 +1,23 @@
+"foo".encode("utf-8") # b"foo"
+"foo".encode("u8") # b"foo"
+"foo".encode() # b"foo"
+
+string = "hello there"
+string.encode("utf-8") # string.encode()
+
+"""
+Lorem
+
+Ipsum
+""".encode("utf-8") # b"\nLorem\n\nIpsum\n"
+
+"unicode text©".encode("utf-8") # "unicode text©".encode()
+"unicode text©".encode() # "unicode text©".encode()
+
+r"fo\o".encode("utf-8") # b"fo\\o"
```

Comment:
pyupgrade would preserve the raw string with `br"foo\o"`, which seems nicer.

Reply:
I don't really know how to achieve this, because what the AST gives me is already a "bytes" version of the string.
e.g., if I run this code on a `r"fo\o"`, which should replace it by itself
```rust
generator.unparse_expr(constant, 0)
if let Ok(content) = generator.generate() {
    check.amend(Fix::replacement(
        content,
        expr.location,
        expr.end_location.unwrap(),
    ))
}
```
I get `'fo\\o'`

Reply:
We may have to use LibCST to achieve that, since it looks like RustPython evaluates the raw string.

Reply:
I think this can be achieved with a similar method as W605?

https://github.com/charliermarsh/ruff/blob/bf7bf7aa17b7c864268d8ada1559f0ffe297427c/src/pycodestyle/checks.rs#L266-L272

Reply:
Oh yeah, that could work too, and might be easier?

Reply:
Yeah it works great, thanks!